### ISSUE TITLE
r206 - Android Export fails for only for 'Signed Android App Bundle' and iOS builds using 'minify script' throws "Failed to export"

### ISSUE THREAD

**Issue by vundurty** [Scirra/Construct-3-bugs#4010]
## Problem description
I found two issues
1. Exporting iOS builds using 'minify script' throws "Failed to export"
2. Android Export fails for only for 'Signed Android App Bundle' whereas Signed Android Apk' and debug works fine. Please check the error below

Error: Checking Java JDK and Android SDK versions
ANDROID_SDK_ROOT=~~/androidSDK (recommended setting)
ANDROID_HOME=~~/androidSDK (DEPRECATED)
Using Android SDK: ~~/androidSDK
:wrapper

BUILD SUCCESSFUL in 0s
1 actionable task: 1 executed
Subproject Path: CordovaLib
Subproject Path: app
> Task :app:preBuild UP-TO-DATE
> Task :app:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:preBuild UP-TO-DATE
> Task :CordovaLib:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:packageReleaseRenderscript NO-SOURCE
> Task :app:compileReleaseRenderscript NO-SOURCE
> Task :app:generateReleaseResValues
> Task :app:generateReleaseResources
> Task :CordovaLib:compileReleaseRenderscript
> Task :CordovaLib:generateReleaseResValues
> Task :CordovaLib:generateReleaseResources
> Task :CordovaLib:packageReleaseResources
> Task :app:createReleaseCompatibleScreenManifests
> Task :app:extractDeepLinksRelease
> Task :CordovaLib:extractDeepLinksRelease
> Task :CordovaLib:processReleaseManifest
> Task :app:processReleaseManifest
> Task :CordovaLib:compileReleaseLibraryResources
> Task :CordovaLib:compileReleaseAidl NO-SOURCE
> Task :CordovaLib:generateReleaseBuildConfig
> Task :CordovaLib:parseReleaseLocalResources
> Task :app:mergeReleaseResources
> Task :CordovaLib:generateReleaseRFile
> Task :CordovaLib:javaPreCompileRelease
> Task :CordovaLib:compileReleaseJavaWithJavac
> Task :app:bundleReleaseResources
> Task :CordovaLib:bundleLibRuntimeToJarRelease
> Task :app:checkReleaseDuplicateClasses
> Task :app:compileReleaseAidl NO-SOURCE
> Task :app:generateReleaseBuildConfig
> Task :app:javaPreCompileRelease
> Task :app:processReleaseResources
> Task :CordovaLib:bundleLibCompileToJarRelease

> Task :app:compileReleaseJavaWithJavac

> Task :app:dexBuilderRelease
> Task :app:desugarReleaseFileDependencies
> Task :app:mergeExtDexRelease
> Task :app:mergeReleaseShaders
> Task :app:compileReleaseShaders NO-SOURCE
> Task :app:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:mergeReleaseShaders
> Task :CordovaLib:compileReleaseShaders NO-SOURCE
> Task :CordovaLib:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:packageReleaseAssets
> Task :app:mergeReleaseAssets
> Task :app:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:bundleLibResRelease NO-SOURCE
> Task :app:mergeReleaseJavaResource
> Task :app:mergeReleaseJniLibFolders
> Task :CordovaLib:mergeReleaseJniLibFolders
> Task :CordovaLib:mergeReleaseNativeLibs
> Task :CordovaLib:stripReleaseDebugSymbols NO-SOURCE
> Task :CordovaLib:copyReleaseJniLibsProjectOnly
> Task :app:mergeReleaseNativeLibs
> Task :app:stripReleaseDebugSymbols NO-SOURCE
> Task :app:collectReleaseDependencies
> Task :app:configureReleaseDependencies
> Task :app:parseReleaseIntegrityConfig
> Task :app:mergeDexRelease
> Task :app:buildReleasePreBundle
> Task :app:packageReleaseBundle
> Task :app:signReleaseBundle
> Task :app:bundleRelease

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https:/~~/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 4s
44 actionable tasks: 44 executed
Built the following bundle(s):
~~/app-release.aab

## Attach a .c3p
[Test.zip](https://github.com/Scirra/Construct-3-bugs/files/4870025/Test.zip)

## Steps to reproduce

1. Export the project to cordova android or iOS
2.
3.

## Observed result

It should complete the build

## Expected result

I get an error as mentioned in first question

## More details

**Affected browsers/platforms:** 
I tired it on Windows Edge on Windows laptop

**First affected release:**  r206

## System details

<details><summary>View details</summary>

Error: Checking Java JDK and Android SDK versions
ANDROID_SDK_ROOT=~~/androidSDK (recommended setting)
ANDROID_HOME=~~/androidSDK (DEPRECATED)
Using Android SDK: ~~/androidSDK
:wrapper

BUILD SUCCESSFUL in 0s
1 actionable task: 1 executed
Subproject Path: CordovaLib
Subproject Path: app
> Task :app:preBuild UP-TO-DATE
> Task :app:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:preBuild UP-TO-DATE
> Task :CordovaLib:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:packageReleaseRenderscript NO-SOURCE
> Task :app:compileReleaseRenderscript NO-SOURCE
> Task :app:generateReleaseResValues
> Task :app:generateReleaseResources
> Task :CordovaLib:compileReleaseRenderscript
> Task :CordovaLib:generateReleaseResValues
> Task :CordovaLib:generateReleaseResources
> Task :CordovaLib:packageReleaseResources
> Task :app:createReleaseCompatibleScreenManifests
> Task :app:extractDeepLinksRelease
> Task :CordovaLib:extractDeepLinksRelease
> Task :CordovaLib:processReleaseManifest
> Task :app:processReleaseManifest
> Task :CordovaLib:compileReleaseLibraryResources
> Task :CordovaLib:compileReleaseAidl NO-SOURCE
> Task :CordovaLib:generateReleaseBuildConfig
> Task :CordovaLib:parseReleaseLocalResources
> Task :app:mergeReleaseResources
> Task :CordovaLib:generateReleaseRFile
> Task :CordovaLib:javaPreCompileRelease
> Task :CordovaLib:compileReleaseJavaWithJavac
> Task :app:bundleReleaseResources
> Task :CordovaLib:bundleLibRuntimeToJarRelease
> Task :app:checkReleaseDuplicateClasses
> Task :app:compileReleaseAidl NO-SOURCE
> Task :app:generateReleaseBuildConfig
> Task :app:javaPreCompileRelease
> Task :app:processReleaseResources
> Task :CordovaLib:bundleLibCompileToJarRelease

> Task :app:compileReleaseJavaWithJavac

> Task :app:dexBuilderRelease
> Task :app:desugarReleaseFileDependencies
> Task :app:mergeExtDexRelease
> Task :app:mergeReleaseShaders
> Task :app:compileReleaseShaders NO-SOURCE
> Task :app:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:mergeReleaseShaders
> Task :CordovaLib:compileReleaseShaders NO-SOURCE
> Task :CordovaLib:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:packageReleaseAssets
> Task :app:mergeReleaseAssets
> Task :app:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:bundleLibResRelease NO-SOURCE
> Task :app:mergeReleaseJavaResource
> Task :app:mergeReleaseJniLibFolders
> Task :CordovaLib:mergeReleaseJniLibFolders
> Task :CordovaLib:mergeReleaseNativeLibs
> Task :CordovaLib:stripReleaseDebugSymbols NO-SOURCE
> Task :CordovaLib:copyReleaseJniLibsProjectOnly
> Task :app:mergeReleaseNativeLibs
> Task :app:stripReleaseDebugSymbols NO-SOURCE
> Task :app:collectReleaseDependencies
> Task :app:configureReleaseDependencies
> Task :app:parseReleaseIntegrityConfig
> Task :app:mergeDexRelease
> Task :app:buildReleasePreBundle
> Task :app:packageReleaseBundle
> Task :app:signReleaseBundle
> Task :app:bundleRelease

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https:/~~/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 4s
44 actionable tasks: 44 executed
Built the following bundle(s):
~~/app-release.aab

</details>

**Comment by pinellos** [comment]
Same issue here, the problem is the same if you try to export a non-signed bundle...

Error: Checking Java JDK and Android SDK versions
ANDROID_SDK_ROOT=~~/androidSDK (recommended setting)
ANDROID_HOME=~~/androidSDK (DEPRECATED)
Using Android SDK: ~~/androidSDK
:wrapper

BUILD SUCCESSFUL in 0s
1 actionable task: 1 executed
Subproject Path: CordovaLib
Subproject Path: app
> Task :app:preBuild UP-TO-DATE
> Task :app:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:preBuild UP-TO-DATE
> Task :CordovaLib:preReleaseBuild UP-TO-DATE
> Task :CordovaLib:packageReleaseRenderscript NO-SOURCE
> Task :app:compileReleaseRenderscript NO-SOURCE
> Task :app:generateReleaseResValues
> Task :app:generateReleaseResources
> Task :CordovaLib:compileReleaseRenderscript
> Task :CordovaLib:generateReleaseResValues
> Task :CordovaLib:generateReleaseResources
> Task :CordovaLib:packageReleaseResources
> Task :app:createReleaseCompatibleScreenManifests
> Task :app:extractDeepLinksRelease
> Task :CordovaLib:extractDeepLinksRelease
> Task :CordovaLib:processReleaseManifest
> Task :app:processReleaseManifest
> Task :CordovaLib:compileReleaseLibraryResources
> Task :CordovaLib:compileReleaseAidl NO-SOURCE
> Task :CordovaLib:generateReleaseBuildConfig
> Task :CordovaLib:parseReleaseLocalResources
> Task :CordovaLib:generateReleaseRFile
> Task :CordovaLib:javaPreCompileRelease
> Task :CordovaLib:compileReleaseJavaWithJavac
> Task :app:mergeReleaseResources
> Task :CordovaLib:bundleLibRuntimeToJarRelease
> Task :app:bundleReleaseResources
> Task :app:checkReleaseDuplicateClasses
> Task :app:compileReleaseAidl NO-SOURCE
> Task :app:generateReleaseBuildConfig
> Task :app:javaPreCompileRelease
> Task :app:processReleaseResources
> Task :CordovaLib:bundleLibCompileToJarRelease

> Task :app:compileReleaseJavaWithJavac

> Task :app:dexBuilderRelease
> Task :app:desugarReleaseFileDependencies
> Task :app:mergeExtDexRelease
> Task :app:mergeReleaseShaders
> Task :app:compileReleaseShaders NO-SOURCE
> Task :app:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:mergeReleaseShaders
> Task :CordovaLib:compileReleaseShaders NO-SOURCE
> Task :CordovaLib:generateReleaseAssets UP-TO-DATE
> Task :CordovaLib:packageReleaseAssets
> Task :app:mergeReleaseAssets
> Task :app:mergeDexRelease
> Task :app:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:processReleaseJavaRes NO-SOURCE
> Task :CordovaLib:bundleLibResRelease NO-SOURCE
> Task :app:mergeReleaseJniLibFolders
> Task :app:mergeReleaseJavaResource
> Task :CordovaLib:mergeReleaseJniLibFolders
> Task :CordovaLib:mergeReleaseNativeLibs
> Task :CordovaLib:stripReleaseDebugSymbols NO-SOURCE
> Task :CordovaLib:copyReleaseJniLibsProjectOnly
> Task :app:collectReleaseDependencies
> Task :app:mergeReleaseNativeLibs
> Task :app:stripReleaseDebugSymbols NO-SOURCE
> Task :app:buildReleasePreBundle
> Task :app:configureReleaseDependencies
> Task :app:parseReleaseIntegrityConfig
> Task :app:packageReleaseBundle
> Task :app:signReleaseBundle
> Task :app:bundleRelease

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https:/~~/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 5s
44 actionable tasks: 44 executed
Built the following bundle(s):
~~/app-release.aab

**Comment by AshleyScirra** [comment]
It looks like since we updated to cordova-android@9.0.0, the output filename changed which caused it to think the build failed. The fix was simple, it just needed to look for another output filename. This change is now live on the build server, so it should already be working in r206 now (I just tried both signed & unsigned AAB and both worked), but I'll add it to the changelog for r207 to highlight the fix there. Let me know if you still have any issues.
